src/client.cpp

```diff
@@ -868,6 +868,31 @@ int Client<IoType, SwitchDataIntegrity, SwitchActivityInfo, SwitchCycleDuration,
             if (p_client_bind_addr->ss_family != AF_UNSPEC) {
                 socklen_t client_bind_addr_len = g_pApp->m_const_params.client_bind_info_len;
                 std::string hostport = sockaddr_to_hostport(p_client_bind_addr);
+                struct sockaddr_store_t unix_addr;
+                socklen_t unix_addr_len;
+                if (p_client_bind_addr->ss_family == AF_UNIX && g_pApp->m_const_params.sock_type == SOCK_DGRAM) { // Need to bind localy
+                    if (strlen(p_client_bind_addr->addr_un.sun_path) == 0) { // no specific addr client_info was provoided
+                        log_dbg("No client name was provided, setting addr_un.sun_path to %s_%d_%d\n",s_user_params.addr.addr_un.sun_path,
+                        getpid(),ifd);
+                        std::string sun_path = s_user_params.addr.addr_un.sun_path;
+                        char pid_buffer [MAX_FDS_NUM];
+                        char ifd_buffer [MAX_FDS_NUM];
+                        snprintf(pid_buffer, sizeof(pid_buffer), "%d", getpid());
+                        snprintf(ifd_buffer, sizeof(ifd_buffer), "%d", ifd);
+                        sun_path = sun_path + "_" + pid_buffer + "_" + ifd_buffer;
```

Comment:
I suggest to do something like this:
```suggestion
                        std::string sun_path = sun_path_to_string(s_user_params.addr.addr_un.sun_path) + '_' + std::to_string(getpid()) + '_' + std::to_string(ifd);
```

You can start to use `std::to_string` because C++11 support is coming in PR #185.

`sun_path_to_string` must handle the corner case when sun_path does not include the trailing zero.

Reply:
Your new function `sun_path_to_string` does not match the name (because does not perform conversion from sun_path to std_string inside the function) and new code still lacks handling of absent zero at the end of buffer.